test/python/compiler/test_transpiler.py

```diff
@@ -696,6 +699,37 @@ def test_measure_doesnt_unroll_ms(self, optimization_level):
         self.assertEqual(qc, out)
 
 
+class StreamHandlerRaiseException(StreamHandler):
+    """Handler class that will raise an exception on formatting errors."""
+
+    def handleError(self, record):
+        raise sys.exc_info()
+
+
+class TestLogTranspile(QiskitTestCase):
+    """Testing the log_transpile option."""
+
+    def setUp(self):
+        logger = getLogger()
+        logger.setLevel('DEBUG')
+        self.output = io.StringIO()
+        logger.addHandler(StreamHandlerRaiseException(self.output))
+        self.circuit = QuantumCircuit(QuantumRegister(1))
+
+    def assertTranspileLog(self, log_msg):
+        """ Runs the transpiler and check for logs containing specified message"""
+        transpile(self.circuit)
+        self.output.seek(0)
+        # Filter unrelated log lines
+        output_lines = self.output.readlines()
+        transpile_log_lines = [x for x in output_lines if x.__contains__(log_msg)]
```

Comment:
Is there a reason not to do:

```suggestion
        transpile_log_lines = [x for x in output_lines if log_msg in x]
```
Calling the private methods seems a bit weird.